src/app/views/query-runner/request/permissions/TabList.tsx

```diff
@@ -52,25 +62,27 @@ const TabList = ({ columns, classes, renderItemColumn, renderDetailsHeader, maxH
     return displayNotSignedInMessage();
   }
 
-
+  if (filteredPermissions.length === 0) {
+    return displayNoPermissionsFoundMessage();
+  }
 
   return (
     <>
       <Label className={classes.permissionLength}>
-        <FormattedMessage id='Permissions' />&nbsp;({permissions.length})
+        <FormattedMessage id='Permissions' />&nbsp;({filteredPermissions.length})
       </Label>
       <Label className={classes.permissionText}>
-        {!tokenPresent && <FormattedMessage id='sign in to consent to permissions' />}
-        {tokenPresent && <FormattedMessage id='permissions required to run the query' />}
+        {!tokenPresent && <FormattedMessage id={permissionModeType ? 'sign in to consent to permissions' : 'sign in to consent to application permissions'} />}
+        {tokenPresent && <FormattedMessage id={permissionModeType ? 'permissions required to run the query' : 'application permissions required to run the query'} />}
       </Label>
       <DetailsList styles={{ root: { maxHeight } }}
-        items={permissions}
+        items={filteredPermissions}
         columns={columns}
         onRenderItemColumn={(item?: any, index?: number, column?: IColumn) => renderItemColumn(item, index, column)}
         selectionMode={SelectionMode.none}
         layoutMode={DetailsListLayoutMode.justified}
         onRenderDetailsHeader={(props?: any, defaultRender?: any) => renderDetailsHeader(props, defaultRender)} />
-      {permissions && permissions.length === 0 &&
+      {filteredPermissions && filteredPermissions.length === 0 &&
         displayNoPermissionsFoundMessage()
```

Comment:
```suggestion
      {filteredPermissions 
        && filteredPermissions.length === 0 
        && displayNoPermissionsFoundMessage()
```
This isn't on ya'll at all because this line existed before, but I'm pretty sure the &&'s should be newline-d here for readability.

Reply:
resolved in https://github.com/LokiLabs/microsoft-graph-explorer-v4/pull/16/commits/55666db4d0964581599d5f907b355fb066034d67